starter-code/scripts/models/article.js

```diff
@@ -49,16 +49,31 @@ Article.fetchAll = function() {
     /* When our data is already in localStorage:
     1. We can process and load it,
     2. Then we can render the index page.  */
+    var ipsum = JSON.parse(localStorage.getItem('hackerIpsum'));
+    // for (var i of ipsum) {
+    //   new Article(i.keys);
+    //   console.log('the if has executed');
+    // }
+    articleView.renderIndexPage();
   } else {
     /* Without our localStorage in memory, we need to:
     1. Retrieve our JSON file with $.getJSON
       1.a Load our json data
       1.b Store that data in localStorage so that we can skip the server call next time,
       1.c And then render the index page.*/
-  }
-};
-
-
+      $.getJSON('../../data/hackerIpsum.json1').then(
+      function(data) {
+        console.log(data);
+      },
+      function() {
+        //console.error('This did not work');
+        console.log('This did not work');
+        }
+      );
+      articleView.renderIndexPage();
+      console.log('the else has executed');
```

Comment:
One way you could refactor it is by doing all of your data processing inside of the first callback function passed to the 'then' block.